WolvenKit.App/Helpers/ProjectResourceTools.cs

```diff
@@ -358,141 +361,208 @@ public async Task MoveAndRefactor(Cp77Project activeProject, string sourcePath,
-                if (sourceInRaw != sourceAbsPath && Directory.Exists(sourceInRaw))
-                {
-                    FileHelper.MoveRecursively(sourceInRaw, destAbsPath.Replace("archive", "raw"), overwriteFiles, projectRootPath,
-                        _loggerService);
-                }
+        // Delete only files that were successfully replaced
+        var successfulReplacements = fileReplacements.Where((kvp) => File.Exists(kvp.Value)).ToDictionary();
 
-                _loggerService.Info($"Moved \"{sourceRelPath}{Path.DirectorySeparatorChar}*\" to \"{destRelPath}\"");
-            }
+        foreach (var originalFile in successfulReplacements.Keys.Where(File.Exists))
+        {
+            File.Delete(originalFile);
+        }
 
-            catch (Exception e)
-            {
-                _loggerService.Error(
-                    $"Error when trying to move \"{sourceRelPath}{Path.DirectorySeparatorChar}*\": {e.Message}");
-            }
+        // Delete empty directories
+        if (successfulReplacements.Count > 0)
+        {
+            DeleteEmptyDirectoriesRecursive(sourceFileOrDirAbsPath);
         }
 
         if (!refactor)
         {
             return;
         }
 
-     
-        var newRelPath = activeProject.GetResourcePathFromRoot(destAbsPath);
+        ReplacePathInProject(activeProject, successfulReplacements);
+    }
+
+    private static void MoveDirectoryAndChildren(string sourceAbsPath, string destAbsPath)
+    {
+        if (!Directory.Exists(sourceAbsPath))
+        {
+            return;
+        }
 
-        ReplacePathInProject(activeProject, sourceAbsPath, newRelPath);
+        if (!Directory.Exists(destAbsPath))
+        {
+            Directory.CreateDirectory(destAbsPath);
+        }
+
+        foreach (var directory in Directory.GetDirectories(sourceAbsPath, "*", SearchOption.AllDirectories))
+        {
+            var relativePath = Path.GetRelativePath(sourceAbsPath, directory);
+            var targetDirPath = Path.Combine(destAbsPath, relativePath);
+
+            Directory.CreateDirectory(targetDirPath);
+        }
+
+        Directory.Delete(sourceAbsPath, true);
+    }
+
+    private static void DeleteEmptyDirectoriesRecursive(string directoryPath)
+    {
+        if (!Directory.Exists(directoryPath) ||
+            Directory.GetFiles(directoryPath, "*", SearchOption.AllDirectories).Length != 0)
+        {
+            return;
+        }
+
+        foreach (var subDirectory in Directory.GetDirectories(directoryPath, "*", SearchOption.AllDirectories))
+        {
+            DeleteEmptyDirectoriesRecursive(subDirectory);
+        }
+
+        if (Directory.GetDirectories(directoryPath, "*", SearchOption.AllDirectories).Length == 0)
+        {
+            Directory.Delete(directoryPath, true); // Delete the root directory
+        }
+    }
+
+    private Task MoveFileAsync(string sourcePath, string targetPath, Cp77Project activeProject, bool isDirectory)
+    {
+        try
```

Comment:
couldn't we just use FileInfo.MoveTo ?

Reply:
That caused other issues in the past, which is why this function exists :/